browser/brave_wallet/brave_wallet_tab_helper.cc

```diff
@@ -107,11 +107,24 @@ void BraveWalletTabHelper::BindSolanaProvider(
   tab_helper->solana_provider_receivers_.Add(
       std::make_unique<SolanaProviderImpl>(
           *host_content_settings_map, brave_wallet_service,
-          std::make_unique<BraveWalletProviderDelegateImpl>(web_contents,
-                                                            frame_host)),
+          std::make_unique<BraveWalletProviderDelegateImpl>(
+              web_contents, frame_host->GetGlobalId())),
       std::move(receiver));
 }
 
+base::RepeatingCallback<std::unique_ptr<BraveWalletProviderDelegate>()>
+BraveWalletTabHelper::CreateDelegateFactory(
+    content::GlobalRenderFrameHostId host_id) {
+  return base::BindRepeating(&BraveWalletTabHelper::CreateDelegate,
+                             base::Unretained(this), host_id);
+}
+
```

Comment:
LEt's either implement `BraveWalletProviderDelegate::Clone()` or bind to CreateDelegate being a free function